src/main/scala/org/constellation/gossip/sampling/RandomPeerSampling.scala

```diff
@@ -0,0 +1,37 @@
+package org.constellation.gossip.sampling
+
+import cats.effect.Concurrent
+import cats.implicits._
+import org.constellation.p2p.Cluster
+import org.constellation.schema.{Id, NodeState}
+
+import scala.collection.immutable.Seq._
+import scala.util.Random
+
+/**
+  * Peer sampling strategy based on uniform random sample selection
+  */
+class RandomPeerSampling[F[_]: Concurrent](selfId: Id, cluster: Cluster[F]) extends PeerSampling[F, Id] {
+
+  def selectPaths(fanout: Int): F[Seq[IndexedSeq[Id]]] =
+    getPeers.map(randomPath(_, fanout))
+
+  private def randomPath(peers: Set[Id], fanout: Int): Seq[IndexedSeq[Id]] = {
+    def partition(list: IndexedSeq[Id], count: Int): List[IndexedSeq[Id]] = {
+      val size = list.size / count
+      if (count > 1)
+        list.take(size) :: partition(list.drop(size), count - 1)
+      else
+        list :: Nil
+    }
+
+    partition(Random.shuffle(peers.toIndexedSeq), fanout)
+      .map(selfId +: _ :+ selfId)
+      .map(IndexedSeq[Id](_: _*))
+  }
+
+  private def getPeers: F[Set[Id]] =
+    cluster.getPeerInfo
+      .map(_.filter { case (_, pd) => NodeState.isNotOffline(pd.peerMetadata.nodeState) }.keySet)
```

Comment:
Maybe with the push strategy we should not be sending data to a leaving peer?

Reply:
fetching peer proposals uses only `isNotOffline` check and it seems to be working 🤔 but you may be right. We need to test that.